File: backend/tests/test_constants.py
"""Test constants for credentials and test data.

This file centralizes all test credentials to avoid hardcoding them throughout test files.
These are ONLY for testing purposes and should NEVER be used in production.
"""
import os
import secrets
import string


def generate_test_password(length: int = 12) -> str:
    """Generate a random test password."""
    alphabet = string.ascii_letters + string.digits + string.punctuation
    return ''.join(secrets.choice(alphabet) for _ in range(length))


# Test user credentials
class TestCredentials:
    """Centralized test credentials."""

    # Basic test user
    TEST_EMAIL = "test@example.com"
    TEST_USERNAME = "testuser"
    TEST_PASSWORD = os.getenv("TEST_PASSWORD", "TestPass123!")

    # Admin test user
    ADMIN_EMAIL = "admin@test.com"
    ADMIN_USERNAME = "adminuser"
    ADMIN_PASSWORD = os.getenv("TEST_ADMIN_PASSWORD", "AdminPass123!")

    # Alternative test users for specific scenarios
    TEST_EMAIL_2 = "test2@example.com"
    TEST_USERNAME_2 = "testuser2"

    # Duplicate testing
    DUPLICATE_EMAIL = "duplicate@example.com"
    DUPLICATE_USERNAME = "duplicateuser"

    # Login testing
    LOGIN_EMAIL = "login@example.com"
    LOGIN_USERNAME = "loginuser"

    # Inactive user testing
    INACTIVE_EMAIL = "inactive@example.com"
    INACTIVE_USERNAME = "inactiveuser"

    # Current user testing
    CURRENT_EMAIL = "current@example.com"
    CURRENT_USERNAME = "currentuser"

    # Flow testing
    FLOW_EMAIL = "flow@example.com"
    FLOW_USERNAME = "flowuser"

    # Wrong password for testing
    WRONG_PASSWORD = "WrongPass123!"

    # Invalid email for testing
    INVALID_EMAIL = "not-an-email"


# Database credentials (for documentation only - never hardcoded in actual config)
class DatabaseCredentials:
    """Database credentials - should only be used in documentation examples."""

    # Example PostgreSQL connection (DO NOT USE IN PRODUCTION)
    EXAMPLE_DB_USER = "pastryjoy_user"
    EXAMPLE_DB_PASSWORD = "use_strong_password_here"
    EXAMPLE_DB_HOST = "localhost"
    EXAMPLE_DB_PORT = "5432"
    EXAMPLE_DB_NAME = "pastryjoy_db"

    @staticmethod
    def get_example_connection_string() -> str:
        """Get an example connection string for documentation."""
        return (
            f"postgresql+asyncpg://{DatabaseCredentials.EXAMPLE_DB_USER}:"
            f"{DatabaseCredentials.EXAMPLE_DB_PASSWORD}@"
            f"{DatabaseCredentials.EXAMPLE_DB_HOST}:"
            f"{DatabaseCredentials.EXAMPLE_DB_PORT}/"
            f"{DatabaseCredentials.EXAMPLE_DB_NAME}"
        )


# API Testing
class APITestConstants:
    """Constants for API testing."""

    TEST_ORIGIN = "http://localhost:5173"
    TEST_HOST = "http://test"

    # Token expiration for testing (in seconds)
    TEST_TOKEN_EXPIRATION = 3600  # 1 hour

File: frontend/src/test/testConstants.ts
/**
 * Test constants for credentials and test data.
 *
 * This file centralizes all test credentials to avoid hardcoding them throughout test files.
 * These are ONLY for testing purposes and should NEVER be used in production.
 */

/**
 * Generate a random test password
 */
export const generateTestPassword = (length: number = 12): string => {
  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*';
  let password = '';
  const array = new Uint32Array(length);
  crypto.getRandomValues(array);
  for (let i = 0; i < length; i++) {
    password += chars[array[i] % chars.length];
  }
  return password;
};

/**
 * Test user credentials
 */
export const TestCredentials = {
  // Basic test user
  TEST_EMAIL: 'test@example.com',
  TEST_USERNAME: 'testuser',
  TEST_PASSWORD: process.env.TEST_PASSWORD || 'TestPass123!',

  // Admin test user
  ADMIN_EMAIL: 'admin@test.com',
  ADMIN_USERNAME: 'adminuser',
  ADMIN_PASSWORD: process.env.TEST_ADMIN_PASSWORD || 'AdminPass123!',

  // Alternative test users
  TEST_EMAIL_2: 'test2@example.com',
  TEST_USERNAME_2: 'testuser2',

  // Specific test scenarios
  USER_EMAIL: 'user@example.com',
  NEW_EMAIL: 'new@example.com',
  JOHN_EMAIL: 'john@example.com',

  // Invalid credentials for testing
  WRONG_PASSWORD: 'WrongPass123!',
  INVALID_EMAIL: 'not-an-email',
} as const;

/**
 * Mock user data for testing
 */
export const createMockUser = (overrides?: Partial<{
  id: number;
  email: string;
  username: string;
  full_name: string | null;
  role: 'admin' | 'user';
  is_active: boolean;
  created_at: string;
}>) => ({
  id: 1,
  email: TestCredentials.TEST_EMAIL,
  username: TestCredentials.TEST_USERNAME,
  full_name: 'Test User',
  role: 'user' as const,
  is_active: true,
  created_at: new Date().toISOString(),
  ...overrides,
});

/**
 * Mock admin user data for testing
 */
export const createMockAdminUser = (overrides?: Partial<{
  id: number;
  email: string;
  username: string;
  full_name: string | null;
  role: 'admin' | 'user';
  is_active: boolean;
  created_at: string;
}>) => ({
  id: 1,
  email: TestCredentials.ADMIN_EMAIL,
  username: TestCredentials.ADMIN_USERNAME,
  full_name: 'Admin User',
  role: 'admin' as const,
  is_active: true,
  created_at: new Date().toISOString(),
  ...overrides,
});

/**
 * Mock authentication response
 */
export const createMockAuthResponse = (token: string = 'mock-jwt-token') => ({
  access_token: token,
  token_type: 'bearer',
});

/**
 * API test constants
 */
export const APITestConstants = {
  TEST_ORIGIN: 'http://localhost:5173',
  TEST_API_URL: 'http://localhost:8000',
  MOCK_TOKEN: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.mock.token',
} as const;
